Plugins/Public/alley/PlayerRestrictions.cpp

```diff
@@ -768,31 +768,34 @@ bool UserCmd_Process(uint iClientID, const wstring &wscCmd)
 }
 
 #define IS_CMD(a) !wscCmd.compare(L##a)
-
+#define RIGHT_CHECK(a) if(!(cmds->rights & a)) { cmds->Print(L"ERR No permission\n"); return true; }
 bool ExecuteCommandString_Callback(CCmds* cmds, const wstring &wscCmd)
 {
 	returncode = DEFAULT_RETURNCODE;
 
 	if (IS_CMD("showrestrictions"))
 	{
 		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+
+		RIGHT_CHECK(RIGHT_SUPERADMIN)
+
 		return true;
 	}
 	else if (IS_CMD("generateid"))
 	{
 		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+
+		RIGHT_CHECK(RIGHT_SUPERADMIN)
+
 		AdminCmd_GenerateID(cmds, cmds->ArgStrToEnd(1));
 		return true;
 	}
 	else if (IS_CMD("shiptest"))
 	{
 		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
 		HKPLAYERINFO adminPlyr;
-		if (HkGetPlayerInfo(cmds->GetAdminName(), adminPlyr, false) != HKE_OK)
-		{
-			cmds->Print(L"ERR\n");
-			return true;
-		}
+		
+		RIGHT_CHECK(RIGHT_SUPERADMIN)
```

Comment:
this is actually harmless and I question whether it should be admin-restricted, though I also wonder whether it should really be an FLHook command at all.